File: cucumber-messages/ruby/spec/cucumber/messages/messages_spec.rb
require 'cucumber/messages_pb'
require 'json'

module Cucumber
  module Messages
    describe Messages do
      it "builds a pickle doc string" do
        location = Location.new(line: 10, column: 20)
        pds = PickleDocString.new(
          location: location, 
          contentType: 'text/plain', 
          content: 'some\ncontent\n'
        )
        expect(pds.location.line).to(eq(10))
        expect(pds.location.column).to(eq(20))
        expect(pds.contentType).to(eq('text/plain'))
        expect(pds.content).to(eq('some\ncontent\n'))
      end
    end
  end
end

File: cucumber-messages/dotnet/Cucumber.Messages.Specs/MessagesSpec.cs
using Xunit;
﻿using System.IO;
using static Io.Cucumber.Messages.PickleStepArgument.Types;

namespace Io.Cucumber.Messages.Specs
{
    public class MessagesSpec
    {
        [Fact]
        public void SerializesPickeDocString()
        {
            var pickleDocSring = new PickleDocString
            {
                Location = new Location
                {
                    Line = 10,
                    Column = 20
                },
                ContentType = "text/plain",
                Content = "some\ncontent\n"
            };

            byte[] serializedBytes;
            using (MemoryStream stream = new MemoryStream())
            {
                var codedOutputStream = new Google.Protobuf.CodedOutputStream(stream);
                codedOutputStream.WriteMessage(pickleDocSring);
                codedOutputStream.Flush();
                serializedBytes = stream.ToArray();
            }

            PickleDocString parsedCopy = PickleDocString.Parser.ParseDelimitedFrom(new MemoryStream(serializedBytes));
            Assert.Equal(pickleDocSring, parsedCopy);
        }
    }
}
